lib/PlacesProvider.js

```diff
@@ -668,6 +669,88 @@ Links.prototype = {
   }),
 
   /**
+   * Gets the remote tabs links.
+   *
+   * @param {Integer} options
+   *
+   *
+   * @returns {Promise} Returns a promise with the array of links as payload.
+   */
+  getRemoteTabsLinks:  Task.async(function*(options = {}) {
+    let {limit} = options;
+    if (!limit || limit.options > LINKS_QUERY_LIMIT) {
+      limit = LINKS_QUERY_LIMIT;
+    }
+
+    // We first get the URLs open on other devices from SyncedTabs.
+    let clients = yield SyncedTabs.getTabClients();
+
+    if (clients.length === 0) {
+      // No reason to waste cycles doing anything else.
+      return;
+    }
+
+    // get the URLs with the data we need from all the clients
+    let urls = {};
+    for (let client of clients) {
+      for (let tab of client.tabs) {
+        urls[tab.url] = {
+          deviceName: client.name,
+          deviceIsMobile: client.isMobile,
+          lastUsed: tab.lastUsed
+        };
+      }
+    }
+
```

Comment:
kinda a nit: the nested loops can be re-written more effectively using forEach + map:
```js
let urls = {};
clients.forEach(client => client.tabs.map(tab => {
 urls[tab.url] = {
    deviceName: client.name, 
    deviceIsMobile: client.isMobile, 
    lastUsed: tab.lastUsed
   };
 }));
```